OptimizelySDK/Bucketing/UserProfileUtil.cs

```diff
@@ -0,0 +1,64 @@
+﻿/* 
+ * Copyright 2017, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OptimizelySDK.Bucketing
+{
+    public class UserProfileUtils
+    {
+        /// <summary>
+        /// Validate whether a Dictionary<String, Object> can be transformed into a UserProfile.
+        /// </summary>
+        /// <param name="map"The map to check.></param>
+        /// <returns>True if the map can be converted into a UserProfile. False if the map cannot be converted.</returns>
+        public static bool IsValidUserProfileMap(Dictionary<string, object> map)
+        {
+            // The Map must contain a value for the user ID and experiment bucket map
+            if (!map.ContainsKey(UserProfileService.USER_ID_KEY) ||
+                !map.ContainsKey(UserProfileService.EXPERIMENT_BUCKET_MAP_KEY))
+                return false;
+
+            // Try and cast the experimentBucketMap value to a typed map
+            var experimentBucketMap = map[UserProfileService.EXPERIMENT_BUCKET_MAP_KEY] as Dictionary<string, Dictionary<string, string>>;
```

Comment:
I have removed this condition & already in the decisionService it is being dealt with in try/catch condition. 
                    Dictionary<string, object> userProfileMap = UserProfileService.Lookup(userId);
                    if (UserProfileUtil.IsValidUserProfileMap(userProfileMap))
                    {
                        userProfile = UserProfileUtil.ConvertMapToUserProfile(userProfileMap);
                        variation = GetStoredVariation(experiment, userProfile);
                        if (variation != null) return variation;